swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/data/BillableUsageRemittanceRepository.java

```diff
@@ -154,14 +154,26 @@ private Specification<BillableUsageRemittanceEntity> buildSearchSpecification(
     if (Objects.nonNull(filter.getSla())) {
       searchCriteria = searchCriteria.and(matchingSla(filter.getSla()));
     }
-
     if (Objects.nonNull(filter.getHardwareMeasurementType())) {
       searchCriteria =
           searchCriteria.and(matchingHardwareMeasurementType(filter.getHardwareMeasurementType()));
     }
+    if (filter.isExcludeFailures()) {
+      searchCriteria = searchCriteria.and(excludesFailures());
+    }
+
     return searchCriteria;
   }
 
+  private Specification<BillableUsageRemittanceEntity> excludesFailures() {
+    return (root, query, builder) ->
+        builder.or(
+            builder.isNull(root.get(BillableUsageRemittanceEntity_.status)),
+            builder.notEqual(
+                root.get(BillableUsageRemittanceEntity_.status), RemittanceStatus.FAILED),
+            builder.isNotNull(root.get(BillableUsageRemittanceEntity_.retryAfter)));
```

Comment:
I think we should remove the condition on retry_after being populated. In this case we expect the remittance to go through eventually and we'd want to include that in the total.

```suggestion
        builder.or(
            builder.isNull(root.get(BillableUsageRemittanceEntity_.status)),
            builder.notEqual(
                root.get(BillableUsageRemittanceEntity_.status), RemittanceStatus.FAILED))
        );
```

Reply:
This was explicitly asked in the JIRA description: 
"The remittance records should be filtered to exclude hard failures - where status=failed and retryAfter is null (transient failures, where status=failed and retryAfter is non-null should be included)."
Can you clarify this point with @kahowell ? I will schedule a quick meeting for today.

Reply:
@Sgitario I misunderstood the logic in this. This is doing what I think it should be doing and what the card says. Sorry for the confusion